Makefile

```diff
@@ -0,0 +1,206 @@
+.PHONY: help
+help: ## Show this help message
+	@echo 'Usage: make [target]'
+	@echo ''
+	@echo 'Available targets:'
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+
+# Variables
+FRONTEND_REF ?= local/dalec/frontend
+DALEC_DISABLE_DIFF_MERGE ?= 0
+TIMEOUT ?= 10m
+INTEGRATION_TIMEOUT ?= 59m
+GO ?= go
+DOCKER ?= docker
+
+# Build output directory
+BUILD_DIR ?= _output
+BIN_DIR ?= $(BUILD_DIR)/bin
+
+# Go build flags
+GO_BUILD_FLAGS ?= -v
+GO_TEST_FLAGS ?= -v
+
+# Integration test suite (can be overridden: make test-integration SUITE=Mariner2)
+SUITE ?=
+
+##@ Development
+
+.PHONY: generate
+generate: ## Generate required source files
+	$(GO) generate ./...
+
+.PHONY: lint
+lint: ## Run custom linters
+	$(GO) run ./cmd/lint ./...
+
+.PHONY: lint-all
+lint-all: ## Run golangci-lint and custom linters
+	@if command -v golangci-lint >/dev/null 2>&1; then \
+		golangci-lint run --timeout=30m; \
+	else \
+		echo "golangci-lint not found. Install it or use 'make docker-lint'"; \
+		echo "See: https://golangci-lint.run/usage/install/"; \
+		exit 1; \
+	fi
+	$(GO) run ./cmd/lint ./...
+
+.PHONY: fmt
+fmt: ## Format Go code
+	gofmt -w -s .
+
+##@ Build
+
+.PHONY: build
+build: build-frontend build-redirectio ## Build all binaries
+
+.PHONY: build-frontend
+build-frontend: ## Build frontend binary
+	@mkdir -p $(BIN_DIR)
+	$(GO) build $(GO_BUILD_FLAGS) -o $(BIN_DIR)/frontend ./cmd/frontend
+
+.PHONY: build-redirectio
+build-redirectio: ## Build dalec-redirectio binary
```

Comment:
This is not intended to be consumed except by the frontend itself and is also planned to go away (#773)